crates/matrix-sdk/src/encryption/mod.rs

```diff
@@ -321,6 +324,12 @@ impl Client {
         let ((thumbnail_source, thumbnail_info), file) =
             try_join(upload_thumbnail, upload_attachment).await?;
 
+        let filename = filename.to_owned();
+        let (body, filename) = match caption {
+            Some(caption) => (caption, Some(filename)),
+            None => (filename, None),
+        };
```

Comment:
Uh, can you add a code comment explaining what this is about, as it surprised me, please? (It's likely something from the spec I don't know about :-))

(Or even make a helper out of this, since this is done in a few places it seems)

Reply:
I added a comment and a link to the MSC. It can be a bit confusing, yes :P